src/hotspot/share/prims/jvmtiTagMap.cpp

```diff
@@ -944,6 +944,7 @@ class IterateOverHeapObjectClosure: public ObjectClosure {
 
 // invoked for each object in the heap
 void IterateOverHeapObjectClosure::do_object(oop o) {
+  assert(o != nullptr, "Parameter 'o' must not be null!");
```

Comment:
```suggestion
  assert(o != nullptr, "Heap iteration should never produce null");
```
Same with the other assertion please. Though as @plummercj  states I don't see how this will help with the static analysis tool.